### ISSUE TITLE
PDF fails to download with script error Error: Object doesn't support property or method 'Symbol(Symbol.iterator)_b.tqbfeb19a2a

### ISSUE THREAD

**Issue by ReubenChapman** [mozilla/pdf.js#10896]
Attach (recommended) or Link to PDF file here: 

Configuration:
- Web browser and its version:  11.472.17134.0 -> 11.0.101 KB4483187
- Operating system and its version: Windows10 latest fix pack
- PDF.js version: v2.1.147 (build: 1b84b2e)
- Is a browser extension: No

Steps to reproduce the problem:
We cannot reproduce except on client system
1. Load document using Chrome - no issue, PDF renders correctly
2. Load same document using IE11, PDF never renders

What is the expected behavior? (add screenshot)
Document to load and render

What went wrong? (add screenshot)
Grey screen is seen - viewer loads but document never does
There was an error thrown in the browser dev tools log:
Script error:
Line: 1
Error: Object doesn't support property or method 'Symbol(Symbol.iterator)_b.tqbfeb19a2a'

Possible Unhandled Promise Rejection: TypeError: Object doesn't support property or method 'findIndex'

![image](https://user-images.githubusercontent.com/49719956/59200139-7caff000-8b8f-11e9-87c9-fe8aaf37e8e4.png)

From looking at the code in PDF.js the file "fontloader.js" uses the 'for loop' syntax that is not supported in IE.
   for (const font of fonts)
   {
   }

PDF.js has fixed similar issues. most recently https://github.com/mozilla/pdf.js/issues/10401 where this syntax was removed.

We would want to see if PDF.js can change all instances of this in their font handling.

Link to a viewer (if hosted on a site other than mozilla.github.io/pdf.js or as Firefox/Chrome extension): Customer provided a link:
https://eamsdmprep.dwc.ca.gov/eamsnavigator/?desktop=ibm
Go to link, load browser dev tools
Log in with: 
username/password: ibmuser/ibmuser
Double click on "DaejaViewONEIssues"
Double click on "TextSearchable.pdf"
Viewer loads
Attempts to load PDF document but you can see, the entiree file never gets through

Repeat on Chrome, everything works as expected

**Comment by Snuffleupagus** [comment]
> Web browser and its version:  11.472.17134.0 -> 11.0.101 KB4483187

Assuming that this is IE11, which isn't clearly mentioned here, the version numbers are quite confusing; e.g. what's the arrow all about?

Please make sure that you're using the *latest* version of IE11, since nothing else is supported.
Note also that support for IE11 in the PDF.js project is very limited, as outlined in https://github.com/mozilla/pdf.js/wiki/Frequently-Asked-Questions#faq-support, and it will be completely removed in the future.

> PDF.js version: v2.1.147 (build: [1b84b2e](https://github.com/mozilla/pdf.js/commit/1b84b2ed603a18974c701381f67ddca4b392f569))

That does not correspond to an official release, as found at https://github.com/mozilla/pdf.js/releases, and is consequently not supported.

Downloading this PDF file and then opening it with the latest version of the PDF.js default viewer, see https://mozilla.github.io/pdf.js/web/viewer.html, works for me in IE11.
This suggests that there may be an integration bug in your custom implementation, which is going to be *very* difficult for anyone here to assist with without access to a *reduced* test-case.

> PDF.js has fixed similar issues. most recently #10401 where this syntax was removed.
>
> We would want to see if PDF.js can change all instances of this in their font handling.

That was a special case, regarding iteration of `TypedArray`s and thus not relevant here. You're basically asking to remove all occurrences of `for ... of` loops, which is a definite NO at this point in time.
Furthermore, the only `for ... of` loop in `src/display/font_loader.js` was removed in commit 13230a1123c3c540527cef46c3727bd0d63f6294.

*To summarize:* Since this works in the default viewer, in IE11, there doesn't appear to be anything actionable here unfortunately.

**Comment by ReubenChapman** [comment]
Thanks Jonas for the fast response and appreciate all the points you made, 
I suspect we will look to analyse and figure this out in-house



Reuben Chapman
Client Success Engineer EMEA
IBM Analytic Platform, ECM & ILG EMEA
reuben.chapman@uk.ibm.com
+441908354479




ECM Software Services
Product & Lifecyle Notifications
ECM End Of Service (EOS)
IBM Training Worldwide







IBM Ireland Product Distribution Limited registered in Ireland with number 
92815. Registered office: Oldbrook House, 24-32 Pembroke Road, 
Ballsbridge, Dublin 4
